cmd/ipfs/daemon.go

```diff
@@ -408,14 +409,36 @@ func daemonFunc(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment
 		}
 	}
 
+	// Check limits to see if we have enough to run DHT as a server.
+	var sysConnInbound int
+	var sysStreamInbound int
+	var dhtStreamsInbound int
+	if cfg.Swarm.ResourceMgr.Limits != nil {
+		sysConnInbound = cfg.Swarm.ResourceMgr.Limits.System.ConnsInbound
+		sysStreamInbound = cfg.Swarm.ResourceMgr.Limits.System.StreamsInbound
+		dhtLimits, ok := cfg.Swarm.ResourceMgr.Limits.Protocol[dht.ProtocolDHT]
+		if ok {
+			dhtStreamsInbound = dhtLimits.StreamsInbound
+		}
+	}
+
 	switch routingOption {
 	case routingOptionSupernodeKwd:
 		return errors.New("supernode routing was never fully implemented and has been removed")
 	case routingOptionDefaultKwd, routingOptionAutoKwd:
+		mode := dht.ModeAuto
+		if (sysConnInbound != 0 && sysConnInbound <= config.DefaultResourceMgrMinInboundConns) ||
+			(sysStreamInbound != 0 && sysStreamInbound <= config.DefaultResourceMgrMinInboundConns) ||
+			(dhtStreamsInbound != 0 && dhtStreamsInbound <= config.DefaultResourceMgrMinInboundConns) {
```

Comment:
Actually this code is incompatible with https://github.com/ipfs/kubo/issues/9564
Please use control flow, or `bool` to not do this check if unset (instead of a zero).

Reply:
@Jorropo Can you elaborate?

Reply:
Can't really do better without libp2p/go-libp2p#2000 yet.